src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs

```diff
@@ -0,0 +1,197 @@
+                        catch (ArgumentNullException e)
+                        {
+                            _logger.LogError(e, $"Failed to create SHA comparison link for dependency {dep.To.Name} during asset update for subscription {update.SubscriptionId}");
+                        }
+                        shaRangeToLinkId.Add((dep.From.Commit, dep.To.Commit), _startingReferenceId + changesLinks.Count);
+                        changesLinks.Add(changesUri);
+                    }
+                    subscriptionSection.AppendLine($"  - **{dep.To.Name}**: [from {dep.From.Version} to {dep.To.Version}][{shaRangeToLinkId[(dep.From.Commit, dep.To.Commit)]}]");
+                }
+
+                subscriptionSection.AppendLine();
+                for (int i = 0; i < changesLinks.Count; i++)
+                {
+                    subscriptionSection.AppendLine($"[{i + _startingReferenceId}]: {changesLinks[i]}");
+                }
+
+                subscriptionSection.AppendLine();
+                subscriptionSection.AppendLine(DependencyUpdateEnd);
+                subscriptionSection.AppendLine();
+                UpdatePRDescriptionDueConfigFiles(committedFiles, subscriptionSection);
+
+                subscriptionSection.AppendLine();
+                subscriptionSection.AppendLine(sectionEndMarker);
+                _description.Insert(sectionStartIndex, subscriptionSection.ToString());
+
+            }
+            _description.AppendLine();
+            _startingReferenceId += changesLinks.Count;
+        }
+
+        private int RemovePRDescriptionSection(string sectionStartMarker, string sectionEndMarker)
+        {
+            string descriptionString = _description.ToString();
+            int sectionStartIndex = descriptionString.IndexOf(sectionStartMarker);
+            int sectionEndIndex = descriptionString.IndexOf(sectionEndMarker);
+
+            if (sectionStartIndex != -1 && sectionEndIndex != -1)
+            {
+                sectionEndIndex += sectionEndMarker.Length;
+                _description.Remove(sectionStartIndex, sectionEndIndex - sectionStartIndex);
+                return sectionStartIndex;
+            }
+            // if either marker is missing, just append at end and don't remove anything
+            // from the description
+            return _description.Length;
+        }
+
+        /// <summary>
+        /// Goes through the description and finds the biggest reference id. This is needed when updating an exsiting PR.
+        /// </summary>
+        /// <returns></returns>
+        private int GetStartingReferenceId()
+        {
+            Regex regex = new Regex("(?<=^\\[)\\d+(?=\\]:.+)", RegexOptions.Multiline);
+            int maxIndex = 0;
+
+            foreach(var match in regex.Matches(_description.ToString()))
+            {
+                int currentIndex = Int32.Parse(match.ToString());
+                maxIndex = (currentIndex > maxIndex) ? currentIndex : maxIndex;
```

Comment:
[nit] I recommend using `Math.Max`, or you can transform this whole foreach into a LINQ expression using `Max`.